ui/packages/components/src/Select/Select.tsx

```diff
@@ -7,42 +7,53 @@ type SelectProps = {
   label?: string;
   isLabelVisible?: boolean;
   children: React.ReactNode;
+  className?: string;
+};
+
+export type Option = {
+  id: string;
+  name: string;
+  disabled?: boolean;
 };
 
 type MultiProps = {
-  onChange: (value: string[]) => void;
-  defaultValue?: string[];
+  onChange: (value: Option[]) => void;
+  defaultValue?: Option[];
   multiple: true;
 };
 
 type SingleProps = {
-  onChange: (value: string) => void;
-  defaultValue?: string;
+  onChange: (value: Option) => void;
+  defaultValue?: Option;
   multiple?: false;
 };
 
+type Props = SelectProps & (MultiProps | SingleProps);
+
 export function Select({
   defaultValue,
   label,
   isLabelVisible = true,
   children,
   onChange,
   multiple,
-}: SelectProps & (MultiProps | SingleProps)) {
+  className,
+}: Props) {
```

Comment:
**suggestion (non-blocking)**: This can be for future, but it might be nice to allow `Option` or it's `id` to be a generic type that can be passed to the `Select` component (ex. `Select<T>(...`). That way there could be less type checking in components that use Select and pass an onChange handler. We might then be able to drop the `if (isFunctionTimeField(option.id)) {...` checking in the other components.